CRAMv3.tex

```diff
@@ -1708,36 +1894,42 @@ \subsubsection*{Introduction}
 
 \subsubsection*{Offset}
 
-Most of the codings listed below encode positive integer numbers. An integer offset 
+Many of the codings listed below encode positive integer numbers. An integer offset 
 value is used to allow any integer numbers and not just positive ones to be encoded. 
 It can also be used for monotonically decreasing distributions with the maximum 
 not equal to zero. For example, given offset is 10 and the value to be encoded 
 is 1, the actually encoded value would be offset+value=11. Then when decoding, 
 the offset would be subtracted from the decoded value. 
 
-\subsubsection*{Beta coding}
+\subsubsection{External: codec ID 1}
 
-\subsubsection*{Definition}
+Can encode types \textit{Byte}, \textit{Integer}.
 
-Beta coding is a most common way to represent numbers in \emph{binary notation}. 
+External coding is simply storage of data verbatim to an external block with a given ID.
+If the type is \textit{Byte} the data is stored as-is, otherwise for types \textit{Integer} and \textit{Long} the data is stored in ITF8 and LTF8 format respectively.
 
+\subsubsection*{Parameters}
 
-\subsubsection*{Examples}
+CRAM format defines the following parameters of external coding: 
 
-\begin{tabular}{|>{\raggedright}p{222pt}|>{\raggedright}p{222pt}|}
-\hline
-\textbf{Number} & \textbf{Codeword}\tabularnewline
-\hline
-0 & 0\tabularnewline
-\hline
-1 & 1\tabularnewline
+\begin{tabular}{|>{\raggedright}p{100pt}|>{\raggedright}p{100pt}|>{\raggedright}p{230pt}|}
 \hline
-2 & 10\tabularnewline
+\textbf{Data type} & \textbf{Name} & \textbf{Comment}
+\tabularnewline
 \hline
-4 & 100\tabularnewline
+itf8 & external id & id of an external block containing the byte stream\tabularnewline
 \hline
 \end{tabular}
 
+\subsubsection{Beta coding: codec ID 6}
```

Comment:
Curious about the ordering: why not in ID order?

Reply:
I've no idea to be honest!  I didn't reorder anything, so this is as per the original author.  I'll take a look, but I can see a benefit to having golomb-rice immediately next to golomb.

Maybe we can list the recommended ones in order and then the deprecated ones that none of us use (in their own order).   It's arguable whether deprecating them is a change of specification that goes above simple clarification, but the practical side of it is they don't exist in the wild anyway.

Reply:
Fixed reordering.  Also shuffled some other things around in this appendix.